src/cascadia/TerminalApp/App.cpp

```diff
@@ -1248,8 +1249,9 @@ namespace winrt::TerminalApp::implementation
     {
         if (profile.HasIcon())
         {
-            auto path = profile.GetIconPath();
-            winrt::hstring iconPath{ path };
+            std::wstring path{ profile.GetIconPath() };
+            std::wstring envExpandedPath{ wil::ExpandEnvironmentStringsW<std::wstring>(path.data()) };

```

Comment:
I could go either way on this. Initially I had the expansion done in Profile.GetIconPath(); I changed it because I felt it would be inconsistent with other code in Profile. But see my reply to your point [1].